lambda-runtime/src/AWS/Lambda/Client.hs

```diff
@@ -0,0 +1,192 @@
+module AWS.Lambda.Client
+  ( HTTPConfig
+  , LambdaClient
+  , InternalLambdaClientError (..)
+  , LambdaResponse (..)
+  , getHttpConfig
+  , getNextLambdaResponse
+  , sendEventError
+  , sendEventSuccess
+  , sendBootError
+  )
+where
+
+import AWS.Prelude
+import Control.Exception (displayException)
+import Control.Monad.Except (ExceptT(..), liftEither, throwError, withExceptT)
+import System.Environment (lookupEnv)
+
+import qualified Data.Aeson          as JSON
+import qualified Data.ByteString     as BS
+import qualified Network.HTTP.Client as HTTP
+import qualified Network.HTTP.Types  as HTTP
+
+data HTTPConfig = HTTPConfig
+  { request :: HTTP.Request
+  , manager :: HTTP.Manager
+  }
+
+data LambdaResponse = LambdaResponse
+  { event     :: JSON.Value
+  , requestId :: RequestId
+  }
+  deriving stock Show
+
+newtype RequestId = RequestId Text
+  deriving stock (Show)
+
+instance Conversion BS.ByteString RequestId where
+  convert (RequestId requestId) = encodeUtf8 requestId
+
+data InternalLambdaClientError
+  = LambdaContextDecodeError Text
+  | MissingLambdaRunTimeApi
+  | InvalidLambdaRunTimeApi Text
+  | ConnectionError HTTP.HttpException
+  | ResponseBodyDecodeFailure Text
+  | UnExpectedRunTimeError HTTP.Status Text
```

Comment:
@epicallan IMO `UnExpected` feels like fairly uncommon/surprising snake case usage. I'd probably have it as `Unexpected`.

Reply:
Let me disambiguate this Error into the two specific Errors it currently represents such that it doesn't seem like a catch-all snake case constructor.

Reply:
@epicallan similar to [this issue](https://github.com/mbj/mhs/pull/27/files#r701355860), I think he meant this should be `UnexpectedRunTimeError`. since "Unexpected" is a single word.

Reply:
> @epicallan similar to this issue, I think he meant this should be UnexpectedRunTimeError. since "Unexpected" is a single word.

I see, noted. Though, I already split it up into more specific errors. I like I was forced into this direction.